docs/DeveloperGuide.adoc

```diff
@@ -153,94 +153,234 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+image::SearchCommandClassDiagram.svg[]
 
-* **Alternative 1 (current choice):** Use a list to store the history of address book states.
-** Pros: Easy for new Computer Science student undergraduates to understand, who are likely to be the new incoming developers of our project.
-** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `HistoryManager` and `VersionedAddressBook`.
-* **Alternative 2:** Use `HistoryManager` for undo/redo
-** Pros: We do not need to maintain a separate list, and just reuse what is already in the codebase.
-** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
-// end::undoredo[]
+==== Sequence Diagram for Search feature:
 
-// tag::dataencryption[]
-=== [Proposed] Data Encryption
+The sequence diagram for the *Search Command* when searching for any orders that contain *Alice* in any of it's field is shown below:
 
-_{Explain here how the data encryption feature will be implemented}_
+.Search Command Sequence Diagram
+image::SearchCommandSequenceDiagram.svg[]
+
+LogicManager first calls `parseCommand` with arguments representing the user input, `Alice`. The `SearchCommandParser` will then check for any invalid arguments passed by the user. +
+
+[TIP]
+====
+* If the given arguments are valid, `SearchCommandParser` will return  a new `SearchCommand` object. +
+* If the given arguments are invalid or empty, a `ParseException` object will be thrown (not shown in the diagram).
+====
+
+Upon validation of the arguments, `SearchCommandParser` will call for the `tokenize` static method in `ArgumentTokenizer` to get an `ArgumentMultimap` object containing all the user input keywords.
+Each keyword will be stored in the `ArgumentMultimap` object to it's corresponding prefix.
+If no prefix is given, `ArgumentMultimap` will store the user input as a value tagged to an empty string key; the empty string key is referred to as the *preamble*. +
+
+Using the `ArgumentMultimap` object, `SearchCommandParser` will check if the preamble is empty. +
+If the preamble is not empty, a *general search* will be performed in which the `SearchCommandParser` will call the
+`OrderContainsKeywordsPredicate` constructor. The constructor accepts the user input as a parameter (`keyword`). +
+If the preamble is empty, `SearchCommandParser` will call the overloaded constructor of `OrderContainsKeywordsPredicate`, passing the pre-processed user input as `keyword` and the `ArgumentMultimap`. +
+
+For both scenarios, `OrderContainsKeywordsPredicate` object will be returned to `SearchCommandParser`. After which, a new instance of `SearchCommand` object will be created and returned back to the `LogicManager` through `OrderBookParser`.
+
+.Search Command Sequence Diagram 2
+image::SearchCommandSequenceDiagram2.svg[]
```

Comment:
The second arrow for returned result might need to change to from search command to logic manager